src/components/Transaction.tsx

```diff
@@ -1,31 +1,112 @@
 import React, { useEffect, useState } from 'react';
 import { gql } from '../fetch';
-import type { Signature } from '@solana/web3.js';
-import {  } from '@solana/web3.js';
+import type { Signature, Slot, Transaction, TransactionMessage } from '@solana/web3.js';
 
 /**
  * Component GraphQL query.
  */
+
+// Can add more fields based on requirement
 const source = /* GraphQL */ `
+    fragment TransactionMessageAccountKey on TransactionMessageAccountKey {
+        pubkey 
+        signer 
+        source 
+        writable
+    }
     query testQuery($signature: Signature!) {
         transaction(signature: $signature) {
             blockTime
+            message {
+                accountKeys {
+                    ...TransactionMessageAccountKey
+                }
+                addressTableLookups {
+                    accountKey
+                    readableIndexes
+                    writableIndexes
+                }
+                header {
+                    numReadonlySignedAccounts 
+                    numReadonlyUnsignedAccounts 
+                    numRequiredSignatures 
+                }
+                recentBlockhash
+            }
+            meta {
+                fee
+                err
+                logMessages
+                postBalances
+                preBalances
+            }
+            signatures
             slot
+            version
         }
     }
 `;
 
-export default function Transaction() {
-    const [data, setData] = useState();
+/**
+ * Component GraphQL data.
+ */
+type Data = {
+    blockTime: string,
+    // Should Include more type to display info 
+    message: TransactionMessage,
+    // Should have a TransactionMetadata type in sdk
+    meta: {
+        fee: { __bigint: string} | undefined,
```

Comment:
Fields with `__bigint` should never make it into the app. This is just the mechanism I've employed to send `bigint` values over JSON payloads.

If you look at the server, it will serialize `bigint` values using the `__bigint` key, and then the fetchers in the frontend (`src/fetch.ts`) will deserialize them by converting back to `bigint`.